lua/telescope/builtin/__files.lua

```diff
@@ -544,7 +544,20 @@ files.current_buffer_fuzzy_find = function(opts)
               return
             end
 
-            vim.api.nvim_win_set_cursor(0, { selection.lnum, 0 })
+            local searched_for = require("telescope.actions.state").get_current_line()
+            local first_search_char = string.sub(searched_for, 1, 1)
+
+            local column_as_is = string.find(selection.text, first_search_char)
+            local reversed_char
+            if first_search_char == string.upper(first_search_char) then
+              reversed_char = string.lower(first_search_char)
+            else
+              reversed_char = string.upper(first_search_char)
+            end
+            local column_reversed = string.find(selection.text, reversed_char)
+            local column = math.min(column_as_is or math.huge, column_reversed or math.huge)
```

Comment:
sorry can you explain the reason for this upper/lower/reversed stuff? It's not quite obvious to me why it's needed.

Reply:
Sure thing. Ideally I'd want it to jump to first character that gets highlighted from the matched option. I could not figure out how to do that. Instead I am checking what is the leftmost character in the line that it is the same or reversed case and we will jump to it. 

You can search for `fbr` or `Fbr`, note lower and capital case f/F in the beginning. Both will match the same letters on this line

```
local fuzzy_buffer = function(buffer)
      ^     ^    ^
```

So this logic is to account for reversed case of the first character in user's search.

Let me know I managed to to explain this well

Reply:
Hmm.. I think the trouble is that the fuzzy finder scoring and highlighting is dependent of the sorter that's being used.

With `telescope-fzf-native`, `Fbr` will not match `fuzzy_buffer`.
![image](https://github.com/nvim-telescope/telescope.nvim/assets/66286082/e8ca0c98-37f4-4533-9c78-c3f784f63afd)

But with the default telescope sorter (fzy algorithm), it will match.
![image](https://github.com/nvim-telescope/telescope.nvim/assets/66286082/d3cf483a-e283-4d98-97b6-77f9812cf80a)

> Ideally I'd want it to jump to first character that gets highlighted from the matched option

I agree.
Maybe an alternative approach: we can get the highlight positions through the current picker (`action_state.get_current_picker`) with something like `picker.sorter:highlighter(searched_for, selection.ordinal)`.
You can see `Picker:highlight_one_row` in `pickers.lua` for how we apply highlighting to each entry.
This would need an `action_set.select:replace` or something like that however since using the `post` on `enhance`, the picker is already closed.

Reply:
I will play around with it to see if I can get it to work. I will report back once I take a stub at it. Thank you for the code pointers.

Reply:
Thanks for the code pointers @jamestrew ! I have updated the PR. I now has a simpler implementation and more importantly works correctly ie jumps to what was highlighted instead of the first matching character. I have tested it and updated the "after" video in the PR description. I think it is ready to be reviewed and merged if you are happy with the implementation.